src/kernel/thread/thread_sched_wait.c

```diff
@@ -77,6 +86,7 @@ int sched_wait_timeout(clock_t timeout, clock_t *remain) {
 		return res;
 	}
 
+	SCHED_WAIT_TIMER_ADD(tmr);
```

Comment:
It seems you add `tmp` to the list here in `SCHED_WAIT_TIMER_ADD` and above in `SCHED_WAIT_TIMER_INIT`, they both contain `thread_wait_add(&(thread_self()->thread_wait), _tmr)`. It is normal?

Reply:
One of them contain ` thread_wait_add(&(thread_self()->thread_wait_list), &_tmr)`
I want to remove implementation except allocate on stack and register for thread signals, but In the future